File: app/src/proprietary/java/com/blogspot/e_kanivets/moneytracker/util/CrashlyticsProxy.java
package com.blogspot.e_kanivets.moneytracker.util;

import android.content.Context;
import androidx.annotation.Nullable;
import com.google.firebase.analytics.FirebaseAnalytics;

/**
 * Util class that wraps all Crashlytics interactions to disable Answers in
 * Debug mode and allow not including Crashlytics in free (fdroid) builds.
 * Created on 1/11/17.
 *
 * @author Evgenii Kanivets
 */

public class CrashlyticsProxy {
    private static CrashlyticsProxy instance;

    public static CrashlyticsProxy get() {
        if (instance == null) {
            instance = new CrashlyticsProxy();
        }
        return instance;
    }

    private CrashlyticsProxy() {

    }

    private boolean enabled;
    private static FirebaseAnalytics analytics;

    public static void startCrashlytics(Context context) {
        analytics = FirebaseAnalytics.getInstance(context);
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public boolean isEnabled() {
        return enabled;
    }

    public boolean logEvent(@Nullable String eventName) {
        if (enabled) {
            analytics.logEvent(eventName, null);
            return true;
        } else {
            return false;
        }
    }

    public boolean logButton(@Nullable String buttonName) {
        if (enabled) {
            analytics.logEvent(buttonName, null);
            return true;
        } else {
            return false;
        }
    }
}

File: app/src/proprietary/java/com/blogspot/e_kanivets/moneytracker/util/CrashlyticsProxy.kt
package com.blogspot.e_kanivets.moneytracker.util

import android.content.Context
import com.google.firebase.analytics.FirebaseAnalytics

class CrashlyticsProxy private constructor() {

    private var enabled: Boolean = false

    fun setEnabled(enabled: Boolean) {
        this.enabled = enabled
    }

    fun isEnabled(): Boolean = enabled

    fun logEvent(eventName: String?): Boolean {
        return if (enabled) {
            analytics?.logEvent(eventName, null)
            true
        } else {
            false
        }
    }

    fun logButton(buttonName: String?): Boolean {
        return if (enabled) {
            analytics?.logEvent(buttonName, null)
            true
        } else {
            false
        }
    }

    companion object {
        private var instance: CrashlyticsProxy? = null
        private var analytics: FirebaseAnalytics? = null

        @JvmStatic
        fun get(): CrashlyticsProxy {
            if (instance == null) instance = CrashlyticsProxy()
            return instance!!
        }

        @JvmStatic
        fun startCrashlytics(context: Context) {
            analytics = FirebaseAnalytics.getInstance(context)
        }
    }
}
